commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -118,24 +115,117 @@ def update_dockwidget_ids(self):
-        self.dockwidget_controlled_structures.\
-            combobox_input_structure_id.clear()
         db_key = self.dockwidget_controlled_structures.combobox_input_model\
             .currentText()  # name of database
         db = get_database_properties(db_key)
         control_structure = ControlledStructures(
             flavor=db["db_entry"]['db_type'])
         control_structure.start_sqalchemy_engine(db["db_settings"])
-        # Get all id's of the connection nodes
+        # Set the id's of the connection nodes
+        self.dockwidget_controlled_structures.\
+            combobox_input_measuring_point_id.clear()
         list_of_measuring_point_ids = control_structure.get_attributes(
             table_name="v2_connection_nodes", attribute_name="id")
         self.dockwidget_controlled_structures.\
             combobox_input_measuring_point_id.addItems(
                 list_of_measuring_point_ids)
-        # Get the id's of the structures
+        # Set the id's of the structures
+        self.dockwidget_controlled_structures.\
+            combobox_input_structure_id.clear()
         list_of_structure_ids = control_structure.get_attributes(
             table_name="v2_weir_view", attribute_name="weir_id")
         self.dockwidget_controlled_structures.\
             combobox_input_structure_id.addItems(list_of_structure_ids)
+
+    def setup_measuring_station_tab(self):
+        """Setup the measuring station tab."""
+        self.dockwidget_controlled_structures\
+            .pushbutton_input_measuring_point_new.clicked\
+            .connect(self.create_new_measuring_point)
+        tablewidget = self.dockwidget_controlled_structures\
+            .tablewidget_measuring_point
+        start_row = 0
+        tablewidget.setItem(start_row, 0, QTableWidgetItem(""))
+        tablewidget.setCellWidget(
+            start_row, 1, self.dockwidget_controlled_structures
+            .combobox_input_measuring_point_table)
+        tablewidget.setCellWidget(
+            start_row, 2, self.dockwidget_controlled_structures
+            .combobox_input_measuring_point_id)
+        tablewidget.setItem(start_row, 3, QTableWidgetItem(""))
+        tablewidget.setCellWidget(
+            start_row, 3, self.dockwidget_controlled_structures
+            .pushbutton_input_measuring_point_new)
+
+    def create_new_measuring_point(self):
+        """Create a new measuring point."""
+        # Get the model
+        db_key = self.dockwidget_controlled_structures\
+            .combobox_input_model.currentText()
+        db = get_database_properties(db_key)
+        control_structure = ControlledStructures(
+            flavor=db["db_entry"]['db_type'])
+        control_structure.start_sqalchemy_engine(db["db_settings"])
+        # Get last id of measure map or set to 0; set to +1
+        table_name = "v2_control_measure_map"
+        attribute_name = "MAX(id)"
+        try:
+            max_id_measure_map = int(control_structure.get_attributes(
```

Comment:
Getting this error now:

ValueError: invalid literal for int() with base 10: 'None' 
Traceback (most recent call last):
  File "/home/jackieleng/.qgis2/python/plugins/ThreeDiToolbox/commands/Tools/Step 3 - Modify schematisation/control_structures.py", line 174, in create_new_measuring_point
    table_name, attribute_name)[0])
ValueError: invalid literal for int() with base 10: 'None'